src/main/java/seedu/address/logic/parser/AbcliParser.java

```diff
@@ -57,6 +58,8 @@ public static Command parseCommand(String userInput) throws ParseException {
      * @throws ParseException if the mode doesn't exist
      */
     public static void switchMode(ParserMode mode) throws ParseException {
+        requireNonNull(mode);
```

Comment:
Good catch on the missing required values, could have triggered a NullPointerException which can further lead to bugs.